src/operator/tensor/cast_storage-inl.h

```diff
@@ -328,6 +329,53 @@ void CastStorageCsrDnsImpl(const OpContext& ctx,
   });
 }
 
+/*!
+ * \brief Casts a csr matrix to another csr.
+ */
+template<typename xpu>
+void CastStorageCsrCsrImpl(const OpContext& ctx,
+                           const NDArray& csr,
+                           NDArray* output) {
+  mshadow::Stream<xpu>* s = ctx.get_stream<xpu>();
+  if (!csr.storage_initialized()) {
+    FillZerosCsrImpl(s, *output);
+    return;
+  }
+  output->CheckAndAllocAuxData(csr::kIndPtr, csr.aux_shape(csr::kIndPtr));
```

Comment:
nit: `CheckAndAlloc(const std::vector<TShape> &aux_shapes)` will make it neater..